cairo/tests/utils/serde.py

```diff
@@ -274,25 +277,85 @@ def serialize_type(self, path: Tuple[str, ...], ptr) -> Any:
             dict_access_types = get_struct_definition(
                 self.program, dict_access_path
             ).members
-            key_type = dict_access_types["key"].cairo_type
+
+            python_key_type = to_python_type(dict_access_types["key"].cairo_type)
+            # Some mappings have keys that are hashed. In that case, the cairo type name starts with "Hashed".
+            # but in reality, the key is a felt.
+            cairo_key_type = (
+                TypeFelt()
+                if dict_access_types["key"]
+                .cairo_type.scope.path[-1]
+                .startswith("Hashed")
+                else dict_access_types["key"].cairo_type
+            )
+
             value_type = dict_access_types["new_value"].cairo_type
             pointers = self.serialize_pointers(mapping_struct_path, mapping_struct_ptr)
             segment_size = pointers["dict_ptr"] - pointers["dict_ptr_start"]
             dict_ptr = pointers["dict_ptr_start"]
 
-            if origin_cls is set:
-                return {
-                    self._serialize(key_type, dict_ptr + i)
-                    for i in range(0, segment_size, 3)
-                }
-
-            return {
-                self._serialize(key_type, dict_ptr + i): self._serialize(
+            dict_data = {
+                self._serialize(cairo_key_type, dict_ptr + i): self._serialize(
                     value_type, dict_ptr + i + 2
                 )
                 for i in range(0, segment_size, 3)
             }
 
+            serialized_dict = {}
+            tracker_data = self.dict_manager.trackers[dict_ptr.segment_index].data
+            if isinstance(cairo_key_type, TypeFelt):
```

Comment:
A set of felt would match this case no ?

Reply:
no, because the cairo_key_type is a Uint